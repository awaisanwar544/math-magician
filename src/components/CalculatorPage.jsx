import React from 'react';
import Calculator from './Calculator';

const CalculatorPage = () => (
  <div style={{ display: 'flex', justifyContent: 'space-between' }}>
    <h2>Let&apos;s do some Math</h2>
    <Calculator className="calculator" />
  </div>
);

export default CalculatorPage;
