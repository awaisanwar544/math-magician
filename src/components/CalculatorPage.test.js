import React from 'react';
import renderer from 'react-test-renderer';

import CalculatorPage from './CalculatorPage';

describe('Calculator', () => {
  it('Displays correct layout of Calculator', () => {
    const tree = renderer
      .create(<CalculatorPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
