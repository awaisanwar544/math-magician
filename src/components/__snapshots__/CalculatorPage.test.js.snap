// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Displays correct layout of Calculator 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <h2>
    Let's do some Math
  </h2>
  <div
    className="calc-container"
  >
    <div
      className="display-container"
    >
      <p
        className="value"
        id="display-value"
      >
        0
      </p>
    </div>
    <div
      className="keypad"
    >
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="calc-btn operator"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="รท"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="calc-btn operator"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="x"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="calc-btn operator"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="-"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="calc-btn operator"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="+"
      />
      <input
        className="calc-btn other span-2-col"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        className="calc-btn other"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="calc-btn operator"
        data-testid="calc-key"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </div>
</div>
`;
